### PR TITLE
Some additions

### PR DESCRIPTION
**Pull request by franmomu**
`PaymentMethod::$card` -> https://stripe.com/docs/api/payment_methods/object#payment_method_object-card

`Event::$data->previous_attributes` -> https://stripe.com/docs/api/events/object#event_object-data-previous_attributes

`Customer::$invoice_settings` -> https://stripe.com/docs/api/customers/object#customer_object-invoice_settings

`PaymentIntent::$next_action` -> https://stripe.com/docs/api/payment_intents/object#payment_intent_object-next_action

### CODE REVIEW

**Comment by spaze** [comment]
Hi, thanks for the PR! I haven't added some of these yet even though I use them as well (like the `card` and the `previous_attributes` ones) because they're not always available and I'm not sure I want to add them when this extension has no way of saying that.

Because there might be a code where this might come and bite you :-) Say `if ($method->card === null)` when `card` is not available might not be flagged.

I'll try to explore if the option to mark a property as optional can be added somehow.

File: src/Retrofit/PaymentIntentNextAction.php [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?php
+declare(strict_types = 1);
+
+namespace Spaze\PHPStan\Stripe\Retrofit;
+
+/**
+ * @property StripeObject $redirect_to_url
+ * @property 'redirect_to_url'|'use_stripe_sdk'|'alipay_handle_redirect'|'oxxo_display_details'|'verify_with_microdeposits' $type
```
Comment (spaze) [inline src/Retrofit/PaymentIntentNextAction.php:8]: TIL that literals can be [written like this](https://phpstan.org/writing-php-code/phpdoc-types#literals-and-constants), nice!

Comment (franmomu) [reply src/Retrofit/PaymentIntentNextAction.php:8]: if they were defined in constants you could use `Foo::TYPE_*` which is more interesting.

**Comment by franmomu** [comment]
> Hi, thanks for the PR! I haven't added some of these yet even though I use them as well (like the `card` and the `previous_attributes` ones) because they're not always available and I'm not sure I want to add them when this extension has no way of saying that.
> 
> Because there might be a code where this might come and bite you :-) Say `if ($method->card === null)` when `card` is not available might not be flagged.
> 
> I'll try to explore if the option to mark a property as optional can be added somehow.

The `card` property has a `Spaze\PHPStan\Stripe\Retrofit\PaymentMethodCard|null` type, I thought adding `null` would make it optional.

**Comment by spaze** [comment]
Took me a loooooong while but I think what I said above doesn't make much sense, sorry.

The optional properties are already defined in the original Stripe libs like e.g. [the `$card` property](https://github.com/stripe/stripe-php/blob/5fc46f43c743c715cb5edeb7be3383efb7b4bb2e/lib/PaymentMethod.php#L53):
```
 * @property null|\Stripe\StripeObject $card
```

So merging this makes sense. Thanks for the contribution! And sorry it took this long.

**Comment by spaze** [comment]
Now released in [2.4.0](https://github.com/spaze/phpstan-stripe/releases/tag/v2.4.0), thanks again.

**Comment by anthonyryan1** [comment]
Is there something in particular I need to enable to cause phpstan to recognize these `@property` definitions?

I haven't figured out why, but after updating to this version. The correct objects are now identified, but the `@property` declarations are not.

I'm getting a lot of messages like:  `Access to an undefined property Spaze\PHPStan\Stripe\Retrofit\PaymentMethodCard::$brand.`

**Comment by anthonyryan1** [comment]
This looks like a regression specific to PHP 8.2, and the depreciation of dynamic properties. We may need to define the properties within the stub, instead of in the phpdoc block.

**Pull request merged**